File: repeatwise-server/src/main/java/com/repeatwise/enums/SetStatus.java
package com.repeatwise.enums;

public enum SetStatus {
    NOT_STARTED("not_started"),
    LEARNING("learning"),
    REVIEWING("reviewing"),
    MASTERED("mastered");

    private final String value;

    SetStatus(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
} 

File: spaced_learning_app/lib/features/learning_set/domain/entities/set_status.dart
import 'package:freezed_annotation/freezed_annotation.dart';

@JsonEnum(unknownEnumValue: SetStatus.unknown)
enum SetStatus {
  @JsonValue('not_started') notStarted,
  @JsonValue('learning') learning,
  @JsonValue('reviewing') reviewing,
  @JsonValue('mastered') mastered,
  unknown,
}
